mock/py/mockbuild/backend.py

```diff
@@ -787,7 +789,15 @@ def get_command(mode):
                 # on re-doing it
                 mode += ['--noprep']
 
-        self.buildroot.doChroot(get_command(mode),
+        # when we used dynamic_buildrequires, we must let rpmbuild check deps
+        # becasue rpmbuild will call %genearte_buildrequires once again
+        # to actually add the BuildRequires to the SRPM metadata
+        # we need to abort the build if %genearte_buildrequires generates a new
+        # unsatisfied dependency
```

Comment:
I fail to follow this part, can you elaborate on how this can happen?  We tried to resolve the last nosrc.rpm, nothing was installed (and if there was some unresolved build-dep, we would fail before).  What's the reason for generating a different dependency list?

Reply:
The reason is that rpmbuild MUST do it to be able to store the generated BuildRequires into the SRPM metadata. If rpmbuild skipped this step, mock would have already installed them, sot he build would succeed, but the BuildRequires would be lost.

Reply:
The question is not why the step needs to be repeated, to properly follow this paragraph I want to absolutely understand "How come the dep list is different" ... some unwanted side-effects?